starfish/core/imagestack/parser/numpy/__init__.py

```diff
@@ -122,28 +122,19 @@ def get_tile(self, r: int, ch: int, z: int) -> TileData:
         pos_ch = self.index_labels[Axes.CH].index(ch)
         pos_z = self.index_labels[Axes.ZPLANE].index(z)
 
-        selectors: Mapping[str, Any] = {
-            Axes.ROUND.value: r,
-            Axes.CH.value: ch,
-            Axes.ZPLANE.value: z,
-        }
-
         coordinates: MutableMapping[Coordinates, Tuple[Number, Number]] = dict()
 
         if self.coordinates is not None:
             for coordinate_type, min_selector_value, max_selector_value in (
                     (Coordinates.X, PhysicalCoordinateTypes.X_MIN, PhysicalCoordinateTypes.X_MAX),
                     (Coordinates.Y, PhysicalCoordinateTypes.Y_MIN, PhysicalCoordinateTypes.Y_MAX),
-                    (Coordinates.Z, PhysicalCoordinateTypes.Z_MIN, PhysicalCoordinateTypes.Z_MAX),
             ):
-                min_selectors = dict(selectors)
-                min_selectors[PHYSICAL_COORDINATE_DIMENSION] = min_selector_value.value
-                max_selectors = dict(selectors)
-                max_selectors[PHYSICAL_COORDINATE_DIMENSION] = max_selector_value.value
-
                 coordinates[coordinate_type] = (
-                    self.coordinates.loc[min_selectors].item(),
-                    self.coordinates.loc[max_selectors].item())
+                    float(self.coordinates[coordinate_type][0]),
+                    float(self.coordinates[coordinate_type][-1]))
+            # For the z coordinate, take the average.
+            average_z_coord = np.average(self.coordinates[Coordinates.Z])
```

Comment:
If you pass this a full 5D tensor (i.e., where z has many layers) into `from_numpy`, this will mess up your z coordinates.